syntax = "proto3";
package maum.brain.resolution;
service SuperResolutionGRPC{
  rpc Upload(stream FileChunk) returns(UploadStatus){}
  //status(1) : message -> file_key
  //status(2) : upload error

  rpc CheckProcess(KeyMessage) returns(ProcessStatus){}
  //status(6) : Process is in queue
  //status(7) : The file is now processing
  //status(8) : Process completed (can download)
  //status(9) : File is deleted (old file key)
  //status(10) : Wrong Key

  rpc Download(KeyMessage) returns(stream FileChunk){}
  //Wrong key -> empty fileChunk
}
message FileChunk{
  bytes file_bytes = 1;
  //Bytes are divided into MB units. (1MB)
}
message UploadStatus{
  int32 status = 1;
  string file_key = 2;
  string error_msg = 3;
}

message ProcessStatus{
  int32 status = 1;
  string message = 2;
}
message KeyMessage{
  string file_key = 1;
}